pkg/component/etcd/bootstrap/etcd.go

```diff
@@ -264,6 +274,14 @@ func (e *etcdDeployer) Destroy(_ context.Context) error {
 	return nil
 }
 
-func (e *etcdDeployer) emptyPod() *corev1.Pod {
-	return &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "etcd-" + e.values.Role + "-0", Namespace: e.namespace}}
+func (e *etcdDeployer) emptyStatefulSet() *appsv1.StatefulSet {
+	return &appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Name: "etcd-" + e.values.Role + "-0", Namespace: e.namespace}}
```

Comment:
```suggestion
	return &appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Name: "etcd-" + e.values.Role, Namespace: e.namespace}}
```

However, if the suffix is removed some of the replacements `pod.Name` => `statefulSet.Name` will no longer yield the same result.

Reply:
Why should we remove the suffix? I don't know how this all will work with HA etcd clusters, but I guess we will also end up with three `StatefulSet`s for `etcd-{0,1,2}` that we can then translate and push to the various control plane nodes. With only one `StatefulSet`, this approach won't work.
Hence, I would keep it this way for now until this part becomes more clear. WDYT?